### PR TITLE
fix: slack auth model throwing an error

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/10473
Closes: https://github.com/lightdash/lightdash/pull/10716

### Description:

if organization does not have a slack integration it throws an error when visiting integrations page and returns error 500

two routes were affected by this:

`/api/v1/slack/`
`/api/v1/slack/channels`

you can see whole bunch of sentry events here:

https://lightdash.sentry.io/issues/4521767626/events/?project=5959292&sort=-url

THIS PR IS BASED on contributions from @maxdiplogit -> https://github.com/lightdash/lightdash/pull/10716

> This PR deals with improving developer productivity by not relying on network errors on the Frontend and also not requesting /slack/channels endpoint if slack is not enabled.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by ZeRego** [comment]
@IrakliJani do you want to review this external PR that fixes the same problem? 
https://github.com/lightdash/lightdash/pull/10716

**Comment by IrakliJani** [comment]
@ZeRego I'll have a look, yes

File: packages/backend/src/clients/Slack/SlackClient.ts [hunk 1]
```diff
@@ -54,10 +54,16 @@ export class SlackClient {
                 organizationUuid,
             );
 
-        return new WebClient(installation?.token);
+        if (!installation) {
+            throw new Error('Could not find slack installation');
+        }
```
Comment (IrakliJani) [inline packages/backend/src/clients/Slack/SlackClient.ts:59]: the above model call `getInstallationFromOrganizationUuid` was throwing before, so im preserving the behavior here.

File: packages/backend/src/clients/Slack/SlackClient.ts [hunk 2]
```diff
@@ -173,10 +186,17 @@ export class SlackClient {
...
+
+        const installation =
+            await this.slackAuthenticationModel.getInstallationFromOrganizationUuid(
                 organizationUuid,
-            )) || {};
+            );
+
+        if (!installation) {
+            throw new Error('Could not find slack installation');
+        }
```
Comment (IrakliJani) [inline packages/backend/src/clients/Slack/SlackClient.ts:197]: same here. this should throw.

File: packages/backend/src/clients/Slack/Slackbot.ts [hunk 3]
```diff
@@ -189,7 +189,11 @@ export class SlackBot {
                             details?.organizationUuid,
                         );
 
-                    appProfilePhotoUrl = installation?.appProfilePhotoUrl;
+                    if (!installation) {
+                        throw new Error('Could not find slack installation');
+                    }
```
Comment (IrakliJani) [inline packages/backend/src/clients/Slack/Slackbot.ts:194]: same here

**Review by maxdiplogit — APPROVED** [review]
Changes acknowledged and approved 👍

File: packages/backend/src/clients/Slack/SlackClient.ts [hunk 4]
```diff
@@ -54,10 +54,16 @@ export class SlackClient {
                 organizationUuid,
             );
 
-        return new WebClient(installation?.token);
+        if (!installation) {
+            throw new Error('Could not find slack installation');
```
Comment (rephus) [inline packages/backend/src/clients/Slack/SlackClient.ts:58]: Should we use an extended `LightdashError` instead of a generic `Error` ?

**Review by rephus — APPROVED** [review]
Tested locally, works as expected :+1:

**Review by ZeRego — APPROVED** [review]

**Review by rephus — APPROVED** [review]

**Comment by IrakliJani** [comment]
thank you @maxdiplogit for your contribution! 😻

**Comment by maxdiplogit** [comment]
> thank you @maxdiplogit for your contribution! 😻 

Thank you for the support @IrakliJani 😊

**Pull request merged**